File: evals/fixtures/spring-kafka-listener-review/05-missing-error-handler/input.java
// pom.xml: spring-boot 3.3.4, spring-kafka 3.2.4. No CommonErrorHandler / DefaultErrorHandler wired anywhere.
package com.example.audit;

import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Component;

@Component
public class AuditListener {

    private final AuditService audit;

    public AuditListener(AuditService audit) {
        this.audit = audit;
    }

    @KafkaListener(topics = "audit-events", groupId = "audit-svc")
    public void onEvent(ConsumerRecord<String, String> record) {
        // No @Bean CommonErrorHandler exists in this module. Any RuntimeException
        // here falls through to the container's default handler, which logs and
        // drops the record. Poison messages silently disappear from the audit
        // stream — the exact opposite of what "audit" implies.
        audit.record(record.value());
    }
}

File: evals/fixtures/spring-kafka-listener-review/01-unbounded-retry/input.kt
// build.gradle.kts: org.springframework.boot 3.3.4, org.springframework.kafka:spring-kafka 3.2.4
package com.example.orders

import org.apache.kafka.clients.consumer.ConsumerRecord
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.kafka.listener.DefaultErrorHandler
import org.springframework.stereotype.Component
import org.springframework.util.backoff.FixedBackOff

@Configuration
class KafkaErrorHandlerConfig {
    @Bean
    fun defaultErrorHandler(): DefaultErrorHandler {
        // interval=1000ms, maxAttempts defaults to Long.MAX_VALUE → unbounded retry
        return DefaultErrorHandler(FixedBackOff(1000L, Long.MAX_VALUE))
    }
}

@Component
class OrderListener(private val orderService: OrderService) {
    @KafkaListener(topics = ["orders"], groupId = "order-svc")
    fun onOrder(record: ConsumerRecord<String, String>) {
        orderService.process(record.value())
    }
}
